src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -329,6 +328,7 @@ record = _serverStore.Cluster.ReadDatabase(context, Name, out index);
                 if (record == null)
                     DatabaseDoesNotExistException.Throw(Name);
 
+                DatabaseVersion = record!.DatabaseVersion ?? Client.ServerWide.DatabaseVersion.CurrentVersion;
```

Comment:
is this actually true? if DatabaseVersion is null, you will get ThrowDocumentIdTooBigFixVersion here?

Reply:
Shouldn't this be BaseVersion?

Reply:
Yes, because for Restore from backup or snapshot, we will either not get there (`InitializeOptions.SkipLoadingDatabaseRecord`) or set this afterwards. For any other case, it should be the current version